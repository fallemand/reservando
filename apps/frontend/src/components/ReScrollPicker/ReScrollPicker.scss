@import "~@/assets/styles/variables";

.re-scroll-picker {
  width: 100%;
  height: 200px;

  // Overrides
  .center-overlay-custom {
    background-color: rgba(98, 76, 103, 0.1);
    color: $text-primary;
    border-radius: $bdu / 2;
    border: 0;
  }
  .pad-top-overlay-custom {
    background: linear-gradient(0deg ,hsla(0,0%,100%,0.4), $background-primary);
  }

  .pad-bottom-overlay-custom {
    background: linear-gradient(180deg ,hsla(0,0%,100%,0.4), $background-primary);
  }
}
