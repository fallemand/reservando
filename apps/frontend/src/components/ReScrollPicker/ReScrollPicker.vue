<template>
  <ScrollPicker
    class="re-scroll-picker"
    v-bind="{ ...$attrs, modelValue: [$attrs.modelValue], 'onUpdate:modelValue': handleChange }"
    active-class="re-scroll-picker__selected"
  />
</template>

<script lang="ts">
import { defineComponent } from "vue";
import ScrollPicker from "vue3-scroll-picker";

const ReScrollPicker = defineComponent({
  name: "ReScrollPicker",
  components: {
    ScrollPicker,
  },
  inheritAttrs: false,
  methods: {
    handleChange(value) {
      this.$emit("update:modelValue", value[0]);
    },
  },
});

export default ReScrollPicker;
</script>

<style lang="scss" src="./ReScrollPicker.scss" />
