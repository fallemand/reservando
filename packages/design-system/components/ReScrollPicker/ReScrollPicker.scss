@import "../../styles/variables";

.re-scroll-picker {
  width: 100%;
  height: 100%;

  // Overrides
  .column-container {
    width: 100%;
    text-align: center;
  }
  .center-overlay-custom {
    background-color: rgba(98, 76, 103, 0.1);
    color: $text-primary;
    border-radius: $bdu / 2;
    border: 0;
  }
  .pad-top-overlay-custom {
    background: linear-gradient(0deg, hsla(0, 0%, 100%, 0.4), $background-primary);
  }

  .pad-bottom-overlay-custom {
    background: linear-gradient(180deg, hsla(0, 0%, 100%, 0.4), $background-primary);
  }
}
