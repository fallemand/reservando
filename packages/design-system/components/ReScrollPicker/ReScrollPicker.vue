<template>
  <ScrollPicker
    class="re-scroll-picker"
    :options="[options]"
    v-bind="{ ...$attrs, modelValue: [$attrs.modelValue], 'onUpdate:modelValue': handleChange }"
    active-class="re-scroll-picker__selected"
  />
</template>

<script lang="ts">
import { defineComponent } from "vue";
import ScrollPicker from "vue3-scroll-picker";

export interface ReScrollPickerOption {
  label: string;
  value: string | number;
}

const ReScrollPicker = defineComponent({
  name: "ReScrollPicker",
  components: {
    ScrollPicker,
  },
  inheritAttrs: false,
  props: {
    options: {
      type: Array as () => ReScrollPickerOption[],
      required: true,
    },
  },
  methods: {
    handleChange(value: string[] | number[]) {
      this.$emit("update:modelValue", value[0]);
    },
  },
});

export default ReScrollPicker;
</script>

<style lang="scss" src="./ReScrollPicker.scss" />
