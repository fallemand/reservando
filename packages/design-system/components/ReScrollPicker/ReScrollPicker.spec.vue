<script lang="ts">
import { shallowMount } from "@vue/test-utils";
import ReScrollPicker from "./ReScrollPicker.vue";

describe("ReScrollPicker component", () => {
  const props = {
    options: [
      { label: "Test1", value: 1 },
      { label: "Test2", value: 2 },
    ],
  };
  it("should render with default props", () => {
    const wrapper = shallowMount(ReScrollPicker, {
      props,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
</script>
