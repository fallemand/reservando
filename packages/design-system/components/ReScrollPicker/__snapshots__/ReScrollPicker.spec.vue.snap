// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReScrollPicker component should render with default props 1`] = `
<scroll-picker-stub
  active-class="re-scroll-picker__selected"
  class="re-scroll-picker"
  modelvalue=""
  options=""
/>
`;

exports[`ReScrollPicker component should render with posticon 1`] = `
<scroll-picker-stub
  active-class="re-scroll-picker__selected"
  class="re-scroll-picker"
  modelvalue=""
  options=""
/>
`;

exports[`ReScrollPicker component should render with preicon 1`] = `
<scroll-picker-stub
  active-class="re-scroll-picker__selected"
  class="re-scroll-picker"
  modelvalue=""
  options=""
/>
`;
